ibm/resource_ibm_iam_api_key_test.go

```diff
@@ -0,0 +1,156 @@
+
+	"github.com/IBM/platform-services-go-sdk/iamidentityv1"
+)
+
+func TestAccIbmIamApiKeyBasic(t *testing.T) {
+	var conf iamidentityv1.APIKey
+	name := fmt.Sprintf("name_%d", acctest.RandIntRange(10, 100))
+	nameUpdate := fmt.Sprintf("name_%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIbmIamApiKeyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIbmIamApiKeyConfigBasic(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIbmIamApiKeyExists("ibm_iam_api_key.iam_api_key", conf),
+					resource.TestCheckResourceAttr("ibm_iam_api_key.iam_api_key", "name", name),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCheckIbmIamApiKeyConfigBasic(nameUpdate),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("ibm_iam_api_key.iam_api_key", "name", nameUpdate),
+				),
+			},
+		},
+	})
+}
+
+func TestAccIbmIamApiKeyAllArgs(t *testing.T) {
+	var conf iamidentityv1.APIKey
+	name := fmt.Sprintf("name_%d", acctest.RandIntRange(10, 100))
+	description := fmt.Sprintf("description_%d", acctest.RandIntRange(10, 100))
+	storeValue := "false"
+	nameUpdate := fmt.Sprintf("name_%d", acctest.RandIntRange(10, 100))
+	descriptionUpdate := fmt.Sprintf("description_%d", acctest.RandIntRange(10, 100))
+	storeValueUpdate := "true"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIbmIamApiKeyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIbmIamApiKeyConfig(name, description, storeValue),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIbmIamApiKeyExists("ibm_iam_api_key.iam_api_key", conf),
+					resource.TestCheckResourceAttr("ibm_iam_api_key.iam_api_key", "name", name),
+					resource.TestCheckResourceAttr("ibm_iam_api_key.iam_api_key", "description", description),
+					resource.TestCheckResourceAttr("ibm_iam_api_key.iam_api_key", "store_value", storeValue),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCheckIbmIamApiKeyConfig(nameUpdate, descriptionUpdate, storeValueUpdate),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("ibm_iam_api_key.iam_api_key", "name", nameUpdate),
+					resource.TestCheckResourceAttr("ibm_iam_api_key.iam_api_key", "description", descriptionUpdate),
+					resource.TestCheckResourceAttr("ibm_iam_api_key.iam_api_key", "store_value", storeValueUpdate),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "ibm_iam_api_key.iam_api_key",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckIbmIamApiKeyConfigBasic(name string) string {
+	return fmt.Sprintf(`
+
+		resource "ibm_iam_api_key" "iam_api_key" {
+			name = "%s"
```

Comment:
Did the testcase executed because i see "iam_id" is required filed and you have not set it how does it worked?